accessor.go

```diff
@@ -49,12 +45,16 @@ func (sa *ShardAccessor) Shard() shard.Key {
 }
 
 func (sa *ShardAccessor) Blockstore() (ReadBlockstore, error) {
-	bs, err := blockstore.NewReadOnly(sa.data.CarV1Reader(), sa.idx)
+	bs, err := blockstore.NewReadOnly(sa.data, sa.idx)
 	return bs, err
 }
 
 // Close terminates this shard accessor, releasing any resources associated
 // with it, and decrementing internal refcounts.
 func (sa *ShardAccessor) Close() error {
-	return sa.data.Close()
+	if err := sa.data.Close(); err != nil {
+		return fmt.Errorf("failed to close shard: %w", err)
```

Comment:
This error should be logged, but it should not prevent the release from happening.